feat: M2 - Product page improvements

PrimaelQuemerais opened the pull request:
### What
This PR addresses a handful of bugs and introduces some improvements to the product page following the integration of tabs.

Do not merge yet, this contains some sort in progress and I intend to add more fixes from the epic.

- [x] Incorrect tab bar colors are displayed in dark mode.
- [x] Header shows unexpected rounded corners in dark mode.
- [x] #6670
- [x] #6664
- [x] #6662

### Screenshot
<img width="585" height="908" alt="Screenshot 2025-07-21 at 8 49 09 AM" src="https://github.com/user-attachments/assets/b386cf22-7cf2-4d98-9b2b-a068a6429898" />

### Part of 
#6680

PrimaelQuemerais committed:
Refactored product page tabs + bug fixes

PrimaelQuemerais committed:
Refactored product page tabs + bug fixes

PrimaelQuemerais committed:
Refactored product page tabs + bug fixes

PrimaelQuemerais committed:
Refactored product page tabs + bug fixes

PrimaelQuemerais committed:
Merge branch 'openfoodfacts:develop' into product_page

g123k commented:
For the TabBar, I've fixed it in https://github.com/openfoodfacts/smooth-app/pull/6791

PrimaelQuemerais commented:
Thanks @g123k I'll remove my fix from here

PrimaelQuemerais committed:
Reverted tab bar fixes

PrimaelQuemerais committed:
Reverted tab bar fixes

PrimaelQuemerais committed:
Reverted tab bar fixes

PrimaelQuemerais committed:
Reverted tab bar fixes

monsieurtanuki committed:
feat: 6700 - display of "no barcode" data (#6786)

New files:
* `price_category_repository.dart`: Helper for "Category" tag localization.
* `price_category_widget.dart`: Price Category display (no price data here).
* `price_l10n_helper.dart`: Localization helper for prices, managing static cache and downloads.
* `price_label_repository.dart`: Helper for "Labels" tag localization.
* `price_origin_repository.dart`: Helper for "Origins" tag localization.
* `price_taxonomy_repository.dart`: Helper for tag localization using taxonomy.

Impacted files:
* `price_data_value.dart`: now displaying "pricePer" when relevant
* `price_existing_amount_card.dart`: refactored the display of localized tags now using new class `PriceL10nHelper`
* `price_product_widget.dart`: minor refactoring
* `product_prices_list.dart`: now displaying new `PriceCategoryWidget` widget when relevant
* `pubspec.lock`: wtf

monsieurtanuki committed:
feat: 6700 - display of "no barcode" data (#6786)

New files:
* `price_category_repository.dart`: Helper for "Category" tag localization.
* `price_category_widget.dart`: Price Category display (no price data here).
* `price_l10n_helper.dart`: Localization helper for prices, managing static cache and downloads.
* `price_label_repository.dart`: Helper for "Labels" tag localization.
* `price_origin_repository.dart`: Helper for "Origins" tag localization.
* `price_taxonomy_repository.dart`: Helper for tag localization using taxonomy.

Impacted files:
* `price_data_value.dart`: now displaying "pricePer" when relevant
* `price_existing_amount_card.dart`: refactored the display of localized tags now using new class `PriceL10nHelper`
* `price_product_widget.dart`: minor refactoring
* `product_prices_list.dart`: now displaying new `PriceCategoryWidget` widget when relevant
* `pubspec.lock`: wtf

monsieurtanuki committed:
fix: 6785 - no useless progress indicator (#6787)

monsieurtanuki committed:
feat: 6790 - reuse price tokens (#6793)

Impacted files:
* `background_task_add_other_price.dart`: minor refactoring
* `background_task_add_price.dart`: minor refactoring
* `background_task_price.dart`: now using new method `ProductQuery.getPriceToken` to get the token; never closing the token session at the end
* `infinite_scroll_manager.dart`: minor refactoring
* `prices_proofs_page.dart`: minor refactoring
* `product_query.dart`: new method `getPriceToken` that caches the token

dependabot[bot] committed:
chore(deps): bump openfoodfacts in /packages/smooth_app (#6798)

Bumps [openfoodfacts](https://github.com/openfoodfacts/openfoodfacts-dart) from 3.23.0 to 3.24.0.
- [Release notes](https://github.com/openfoodfacts/openfoodfacts-dart/releases)
- [Changelog](https://github.com/openfoodfacts/openfoodfacts-dart/blob/master/CHANGELOG.md)
- [Commits](https://github.com/openfoodfacts/openfoodfacts-dart/compare/v3.23.0...v3.24.0)

---
updated-dependencies:
- dependency-name: openfoodfacts
  dependency-version: 3.24.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump mockito from 5.4.6 to 5.5.0 in /packages/smooth_app (#6797)

Bumps [mockito](https://github.com/dart-lang/mockito) from 5.4.6 to 5.5.0.
- [Release notes](https://github.com/dart-lang/mockito/releases)
- [Changelog](https://github.com/dart-lang/mockito/blob/master/CHANGELOG.md)
- [Commits](https://github.com/dart-lang/mockito/commits)

---
updated-dependencies:
- dependency-name: mockito
  dependency-version: 5.5.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

g123k committed:
Color of the tabbar overflow (#6791)

g123k committed:
Add price: Product's name in the AppBar (#6792)

monsieurtanuki committed:
feat: 6479 - price count as badge for proof images (#6795)

g123k committed:
Revert mobile_scanner to 6.0.10 (#6800)

g123k committed:
fix: Background tasks and support dirs on iOS devices (#6722)

* Fix support dir changed

* Improve code logic

dependabot[bot] committed:
chore(deps): bump l10n_countries in /packages/smooth_app (#6808)

Bumps [l10n_countries](https://github.com/tsinis/sealed_world/tree/main/packages) from 1.1.1 to 1.2.0.
- [Commits](https://github.com/tsinis/sealed_world/commits/l10n_countries-v1.2.0/packages)

---
updated-dependencies:
- dependency-name: l10n_countries
  dependency-version: 1.2.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

g123k committed:
Bump Flutter version to 3.32.8 (#6799)

g123k committed:
Bump Flutter version to 3.32.8 (#6799)

monsieurtanuki committed:
fix: 6784 - readyForPriceTagValidation true only for priceTag and bulk (#6794)

* fix: 6784 - readyForPriceTagValidation true only for priceTag and bulk

Impacted files:
* `background_task_add_price.dart`: added field and parameter `bool bulkProofUpload`; set `readyForPriceTagValidation` as "priceTag and bulk"
* `price_bulk_proof_card.dart`: added warning about AI and image validation
* `price_model.dart`: added field `bool bulkProofUpload`
* `proof_bulk_add_page.dart`: added parameter `bulkProofUpload: true`

* Replaced parameter bulkProofUpload with readyForPriceTagValidation

Impacted files:
* `background_task_add_price.dart`: replaced parameter bulkProofUpload with readyForPriceTagValidation
* `price_bulk_proof_card.dart`: added a switch to change the value of readyForPriceTagValidation
* `price_model.dart`: readyForPriceTagValidation with getter and setter
* `proof_bulk_add_page.dart`: start with the default user value for readyForPriceTagValidation
* `user_preferences.dart`: storing the default user value for readyForPriceTagValidation

* minor fix

* Localization

monsieurtanuki committed:
fix: 6784 - readyForPriceTagValidation true only for priceTag and bulk (#6794)

* fix: 6784 - readyForPriceTagValidation true only for priceTag and bulk

Impacted files:
* `background_task_add_price.dart`: added field and parameter `bool bulkProofUpload`; set `readyForPriceTagValidation` as "priceTag and bulk"
* `price_bulk_proof_card.dart`: added warning about AI and image validation
* `price_model.dart`: added field `bool bulkProofUpload`
* `proof_bulk_add_page.dart`: added parameter `bulkProofUpload: true`

* Replaced parameter bulkProofUpload with readyForPriceTagValidation

Impacted files:
* `background_task_add_price.dart`: replaced parameter bulkProofUpload with readyForPriceTagValidation
* `price_bulk_proof_card.dart`: added a switch to change the value of readyForPriceTagValidation
* `price_model.dart`: readyForPriceTagValidation with getter and setter
* `proof_bulk_add_page.dart`: start with the default user value for readyForPriceTagValidation
* `user_preferences.dart`: storing the default user value for readyForPriceTagValidation

* minor fix

* Localization

teolemon commented:
@PrimaelQuemerais merge conflict

PrimaelQuemerais committed:
Merge branch 'openfoodfacts-develop' into product_page

PrimaelQuemerais committed:
Added localization

PrimaelQuemerais committed:
Added localization

PrimaelQuemerais committed:
Merge branch 'openfoodfacts:develop' into product_page

PrimaelQuemerais commented:
Add a "coming soon" message in the for me tab

PrimaelQuemerais committed:
Merge branch 'openfoodfacts-develop' into product_page

teolemon commented on packages/smooth_app/lib/l10n/app_localizations_am.dart:
```diff
@@ -2536,16 +2541,34 @@ class AppLocalizationsAm extends AppLocalizations {
   @override
   String prices_bulk_proof_upload_step_copying(int index, int count) {
     return 'በአገር ውስጥ ፋይል #$index/$countበመቅዳት ላይ';
+=======
```
merge marks @PrimaelQuemerais

PrimaelQuemerais committed:
Reverted unnecessary changes

PrimaelQuemerais committed:
Fixed issue with translation files

g123k committed:
Merge branch 'develop' into product_page

g123k commented on .devcontainer/devcontainer.json:
```diff
@@ -1,7 +1,7 @@
 // https://github.com/microsoft/vscode-dev-containers/tree/v0.224.3/containers/ubuntu
 {
-    "image": "ghcr.io/cirruslabs/flutter:3.35.0",
+	"image": "ghcr.io/cirruslabs/flutter:3.35.0",
```
Shouldn't be committed

g123k commented on packages/smooth_app/lib/pages/product/summary_card.dart:
```diff
@@ -51,6 +51,7 @@ class SummaryCard extends StatefulWidget {
     this.contentPadding,
     this.buttonPadding,
     this.heroTag,
+    this.roundedCorners = true,
```
Please pass a `BorderRadius` object instead of this `bool`

g123k commented on packages/smooth_app/lib/pages/product/product_page/product_page_tab_controller.dart:
```diff
@@ -0,0 +1,73 @@
+  }
+
+  @override
+  void dispose() {
```
Please move it at the end of the class

g123k commented on packages/smooth_app/lib/pages/product/product_page/product_page_tab_controller.dart:
```diff
@@ -0,0 +1,73 @@
+  bool _isInitialized = false;
+
+  @override
+  void initState() {
```
If empty -> please remove

g123k commented on packages/smooth_app/lib/pages/product/product_page/product_page_tab_controller.dart:
```diff
@@ -0,0 +1,73 @@
+  const ProductPageTabController({required this.product, required this.child});
+
+  final Product product;
+  final Widget Function(List<ProductPageTab> tabs, TabController tabController)
```
This is more a childBuilder, than a child

g123k commented on packages/smooth_app/lib/pages/product/product_page/new_product_page.dart:
```diff
@@ -115,49 +103,68 @@ class ProductPageState extends State<ProductPage>
-            SliverPersistentHeader(
-              delegate: ProductHeaderDelegate(
-                statusBarHeight: MediaQuery.viewPaddingOf(context).top,
+      body: RefreshIndicator(
```
You can remove it (will be done in the EditProductPage)

g123k commented on packages/smooth_app/lib/pages/product/product_page/new_product_page.dart:
```diff
@@ -115,49 +103,68 @@ class ProductPageState extends State<ProductPage>
-      child: _buildTabLayout(hasPendingOperations),
+      child: ProductPageTabController(
+        product: upToDateProduct,
+        child: (List<ProductPageTab> tabs, TabController tabController) {
```
=>

g123k commented on packages/smooth_app/lib/pages/product/product_page/header/product_page_tabs.dart:
```diff
@@ -186,7 +201,24 @@ class ProductPageTabBar extends StatelessWidget {
+          future: _getPricesTotal(product, context),
+          builder: (BuildContext context, AsyncSnapshot<int?> snapshot) {
+            if (!snapshot.hasData || snapshot.data == null) {
+              return const SizedBox.shrink();
```
EMPTY_WIDGET

g123k commented on packages/smooth_app/lib/pages/product/product_page/header/product_page_tabs.dart:
```diff
@@ -186,7 +201,24 @@ class ProductPageTabBar extends StatelessWidget {
+              color: context
+                  .extension<SmoothColorsThemeExtension>()
+                  .secondaryNormal,
+              padding: const EdgeInsets.all(VERY_SMALL_SPACE),
```
EdgeInsetsDirectional

teolemon commented:
@PrimaelQuemerais small merge conflict on localizations

g123k commented:
I have tested the app and here are some issues (by importance):

🔴 Settings with switches are partially clickable
The whole item should be clickable (for now, just the switch)
<img width="1290" height="397" alt="IMG_3358" src="https://github.com/user-attachments/assets/7990f95a-fa51-4357-afb2-cc05deb127d7" />


🟠 Link to the Green-Score guide is incorrect
We now have a dedicated page: https://github.com/openfoodfacts/smooth-app/pull/6887

🟢 `InkWell`s do not follow the shape
<img width="2580" height="776" alt="Untitled 2" src="https://github.com/user-attachments/assets/b1755509-6b13-4118-9c4a-bef55fe63c7b" />


🟢 Visual issue with the collapsed header
🟢 The header should have a shadow when the screen is scrolled
<img width="1290" height="667" alt="IMG_3360" src="https://github.com/user-attachments/assets/e0767e38-5424-4b70-9b25-4332e7128fb6" />


🟢 The search field border should have the same color as the button
🟢 The search field should have a minimal border when not focused
<img width="2580" height="396" alt="Untitled 2" src="https://github.com/user-attachments/assets/793685f0-9de8-4fea-9f3a-9357815df76d" />


🟢 "See all stats": the arrow should be circled
<img width="2580" height="608" alt="Untitled 2" src="https://github.com/user-attachments/assets/7bd3bc1b-ae8f-41a0-81eb-267c440a14fd" />


🟢 Missing icons
<img width="2716" height="1848" alt="1" src="https://github.com/user-attachments/assets/8c577464-1f40-490d-adb4-67376a7381b3" />

teolemon committed:
Merge branch 'develop' into product_page

PrimaelQuemerais commented:
Replaced by https://github.com/openfoodfacts/smooth-app/pull/7037
